Source/WTF/wtf/text/StringCommon.h

```diff
@@ -1240,24 +1260,76 @@ ALWAYS_INLINE bool isNonZeroBulk(simde_uint16x8_t accumulated)
     return simde_vmaxvq_u16(accumulated);
 }
 
+ALWAYS_INLINE uint8_t nonZeroIndexBulk(simde_uint8x16_t value)
+{
+    constexpr simde_uint8x16_t indexMask { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 };
+    return simde_vminvq_u8(simde_vornq_u8(indexMask, value));
+}
+
+ALWAYS_INLINE uint8_t nonZeroIndexBulk(simde_uint16x8_t value)
+{
+    constexpr simde_uint16x8_t indexMask { 0, 1, 2, 3, 4, 5, 6, 7 };
+    return simde_vminvq_u16(simde_vornq_u16(indexMask, value));
+}
+
 template<LChar character, LChar... characters>
-ALWAYS_INLINE simde_uint8x16_t compareBulk(simde_uint8x16_t input)
+ALWAYS_INLINE simde_uint8x16_t equalBulk(simde_uint8x16_t input)
 {
     auto result = simde_vceqq_u8(input, simde_vmovq_n_u8(character));
     if constexpr (!sizeof...(characters))
         return result;
     else
-        return mergeBulk(result, compareBulk<characters...>(input));
+        return mergeBulk(result, equalBulk<characters...>(input));
 }
 
 template<UChar character, UChar... characters>
-ALWAYS_INLINE simde_uint16x8_t compareBulk(simde_uint16x8_t input)
+ALWAYS_INLINE simde_uint16x8_t equalBulk(simde_uint16x8_t input)
 {
     auto result = simde_vceqq_u16(input, simde_vmovq_n_u16(character));
     if constexpr (!sizeof...(characters))
         return result;
     else
-        return mergeBulk(result, compareBulk<characters...>(input));
+        return mergeBulk(result, equalBulk<characters...>(input));
+}
+
+ALWAYS_INLINE simde_uint8x16_t equalBulk(simde_uint8x16_t lhs, simde_uint8x16_t rhs)
+{
+    return simde_vceqq_u8(lhs, rhs);
+}
+
+ALWAYS_INLINE simde_uint16x8_t equalBulk(simde_uint16x8_t lhs, simde_uint16x8_t rhs)
+{
+    return simde_vceqq_u16(lhs, rhs);
+}
+
+template<LChar character, LChar... characters>
+ALWAYS_INLINE simde_uint8x16_t lessThanBulk(simde_uint8x16_t input)
+{
+    auto result = simde_vcltq_u8(input, simde_vmovq_n_u8(character));
+    if constexpr (!sizeof...(characters))
+        return result;
+    else
+        return mergeBulk(result, lessThanBulk<characters...>(input));
+}
```

Comment:
I would have said that I would have done this as 
```c++
template<LChar... characters>
ALWAYS_INLINE simde_uint8x16_t lessThanBulk(simde_uint8x16_t input)
{
    auto result = splatBulk(static_cast<int8_t>(0));
    for (LChar character : { characters... })
        result = mergeBulk(result, simde_vcltq_u8(input, simde_vmovq_n_u8(character)));
    return result;
}
```

but isn't this the same as:

```c++
template<LChar... characters>
ALWAYS_INLINE simde_uint8x16_t lessThanBulk(simde_uint8x16_t input)
{
    auto character = std::max(characters...);
    return simde_vcltq_u8(input, simde_vmovq_n_u8(character));
}
```

Reply:
Since we are not using it right now, removed.